utils/build/docker/dotnet_http/parametric/Endpoints/ApmTestApiOtel.cs

```diff
@@ -0,0 +1,333 @@
+             trace_state = activity.TraceStateString ?? "",
+             remote = activity.HasRemoteParent
+         });
+
+        _logger.LogInformation("OtelSpanContextReturn: {Result}", result);
+        return result;
+    }
+
+    private static async Task OtelSetStatus(HttpRequest request)
+    {
+        var requestBodyObject = await DeserializeRequestObjectAsync(request.Body);
+
+        _logger.LogInformation("OtelSetStatus: {RequestBodyObject}", requestBodyObject);
+
+        var code = requestBodyObject["code"];
+
+        if (Enum.TryParse(code.ToString(), ignoreCase: true, out ActivityStatusCode statusCode))
+        {
+            var activity = FindActivity(requestBodyObject["id"]);
+            activity.SetStatus(statusCode, requestBodyObject["description"].ToString());
+        }
+        else
+        {
+            throw new ApplicationException($"Invalid value for ActivityStatusCode: {code}");
+        }
+
+        _logger.LogInformation("OtelSetStatusReturn");
+    }
+
+    private static async Task OtelSetName(HttpRequest request)
+    {
+        var requestBodyObject = await DeserializeRequestObjectAsync(request.Body);
+
+        _logger.LogInformation("OtelSetName: {RequestBodyObject}", requestBodyObject);
+
+        if (requestBodyObject.TryGetValue("id", out var id))
+        {
+            var activity = FindActivity(id);
+            activity.DisplayName = requestBodyObject["name"].ToString() ?? string.Empty;
+        }
+
+        _logger.LogInformation("OtelSetName");
+    }
+    
+    private static async Task OtelSetAttributes(HttpRequest request)
+    {
+        var requestBodyObject = await DeserializeRequestObjectAsync(request.Body);
+
+        _logger.LogInformation("OtelSetAttributes: {RequestBodyObject}", requestBodyObject);
+
+        SetTag(FindActivity(requestBodyObject["span_id"]), ((Newtonsoft.Json.Linq.JObject?)requestBodyObject["attributes"])?.ToObject<Dictionary<string, object>>());
+
+        _logger?.LogInformation("OtelSetAttributesReturn");
+    }
+
+    private static async Task<string> OtelFlushSpans(HttpRequest request)
+    {
+        var requestBodyObject = await DeserializeRequestObjectAsync(request.Body);
+
+        _logger.LogInformation("OtelFlushSpans: {RequestBodyObject}", requestBodyObject);
+        
+        await FlushSpans();
+        
+        var result = JsonConvert.SerializeObject(new
+        {
+            success = true,
+        });
+
+        _logger.LogInformation("OtelFlushSpansReturn: {result}", result);
+
+        return result;
+
+    }
+
+    private static async Task OtelFlushTraceStats(HttpRequest request)
+    {
+        _logger.LogInformation("OtelFlushTraceStats: {Request}", request);
+
+        await FlushTraceStats();
+
+        _logger.LogInformation("OtelFlushTraceStatsReturn");
+    }
+    
+    // Helper methods:
+    private static async Task<Dictionary<string, object>> DeserializeRequestObjectAsync(Stream requestBody)
```

Comment:
Ah I didn't see this earlier, this helper method does exactly what I was suggesting in the `ApmTestApi.cs` file 😅 You should definitely share this between the two endpoints files